From 08da8e1ca7339f50ba27b700b3ea6121823bf2c3 Mon Sep 17 00:00:00 2001
From: Makarand R Muley <makarand.r.muley@intel.com>
Date: Mon, 14 Dec 2015 21:17:21 +0530
Subject: [PATCH 5/5] FOR_UPSTREAM [VPG]: drm/i915: Prevent HDMI EDID read
 error message

Driver reads EDID when it either receives hotplug interrupt or it is
forced to read EDID. While loading, driver tries to identify monitors
attached to it and does forced EDID read. In Forced EDID read
scenario, driver tries to read EDID irrespective of whether HDMI
monitor is attached or not.

In absence of connected monitor, EDID read returns NULL. This is
expected behavior. This patch ensures NULL check on EDID doesn't print
error message in this case.

Tracked-On: https://jira01.devtools.intel.com/browse/OAM-16631
Signed-off-by: Makarand R Muley <makarand.r.muley@intel.com>
Change-Id: I8bf6f7cd60b5ebca4138daff94350306867e8531
Reviewed-on: https://android.intel.com:443/460059
---
 drivers/gpu/drm/i915/intel_hdmi.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/intel_hdmi.c b/drivers/gpu/drm/i915/intel_hdmi.c
index 03a8b91..d5ae6db 100644
--- a/drivers/gpu/drm/i915/intel_hdmi.c
+++ b/drivers/gpu/drm/i915/intel_hdmi.c
@@ -1124,7 +1124,8 @@ READ_EDID:
 				goto READ_EDID;
 			}
 
-			DRM_ERROR("Get_hdmi cant read edid\n");
+			if (!force)
+				DRM_ERROR("Get_hdmi cant read edid\n");
 			return NULL;
 		}
 
-- 
1.9.1

